track_oracle/core/track_oracle_core_impl.cxx

```diff
@@ -143,6 +143,19 @@ ::track_oracle_core_impl()
   }
 }
 
+vector< field_handle_type >
+track_oracle_core_impl
+::get_all_field_handles() const
+{
+ boost::unique_lock< boost::mutex > lock( this->api_lock );
```

Comment:
Can this be done with std library calls?

Reply:
I strongly suspect so; however, the rest of the track_oracle singleton uses this pattern. I'm happy to submit a followup branch that de-boosts.

Reply:
That would be good. I believe track_oracle is one of the packages that explicitly requires Boost in the CMake though, so at least it's protected. DeBoostificaction should be a separate PR  for sure.